src/worldfactory.cpp

```diff
@@ -257,45 +258,37 @@ bool worldfactory::save_world(WORLDPTR world, bool is_conversion)
         return false;
     }
 
-    std::ofstream fout;
-    const auto savefile = world->world_path + "/" + FILENAMES["worldoptions"];
-
     if (!assure_dir_exist(world->world_path)) {
         DebugLog( D_ERROR, DC_ALL ) << "Unable to create or open world[" << world->world_name <<
                                     "] directory for saving";
         return false;
     }
 
     if (!is_conversion) {
-        fout.exceptions(std::ios::badbit | std::ios::failbit);
-
-        fout.open(savefile.c_str());
-
-        if (!fout.is_open()) {
-            popup( _( "Could not open the world file %s, check file permissions." ), savefile.c_str() );
-            return false;
-        }
-
-        JsonOut jout( fout, true );
+        const auto savefile = world->world_path + "/" + FILENAMES["worldoptions"];
+        const bool saved = write_to_file( savefile, [&]( std::ostream &fout ) {
+            JsonOut jout( fout );
 
-        jout.start_array();
+            jout.start_array();
 
-        for( auto &elem : world->WORLD_OPTIONS ) {
-            if( elem.second.getDefaultText() != "" ) {
-                jout.start_object();
+            for( auto &elem : world->WORLD_OPTIONS ) {
+                if( elem.second.getDefaultText() != "" ) {
```

Comment:
Preexisting code. Doesn't even show up in the diff without white spaces: https://github.com/CleverRaven/Cataclysm-DDA/pull/16426/files?w=1